capstone-project/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -115,6 +153,12 @@ private static void updateMarker(Marker marker) throws EntityNotFoundException {
         // Find the entity of the marker that needs to be updated using its ID
         Key markerEntityKey = KeyFactory.createKey("Marker", marker.getId());
         Entity markerEntity = datastore.get(markerEntityKey);
+        // If markerEntity has userId, the updated marker should have the same userId
+        Optional<String> userId = Optional.empty();
+        if (markerEntity.hasProperty("userId")){
+            userId = Optional.of((String) markerEntity.getProperty("userId"));
+        }
+        marker.setUserId(userId);
```

Comment:
When can that happen?

Reply:
Right now, user doesn't have to sign in to upload a marker